wallet-new/src/Cardano/Wallet/API/V1/Migration.hs

```diff
@@ -108,3 +111,60 @@ instance Migrate V0.SyncProgress V1.SyncProgress where
                 Just nd | _spLocalCD >= nd -> 100
                 Just nd -> round @Double $ (fromIntegral _spLocalCD / fromIntegral nd) * 100.0
         in pure $ V1.mkSyncProgress (fromIntegral percentage)
+
+
+--
+instance Migrate V0.CAccount V1.Account where
+    eitherMigrate V0.CAccount{..} =
+        V1.Account <$> eitherMigrate caId
+                  -- ^ accId
+                  <*> mapM eitherMigrate caAddresses
+                  -- ^ accAddresses
+                  <*> eitherMigrate caAmount
+                  -- ^ accAmount
+                  <*> pure (V0.caName caMeta)
+                  -- ^ accName
+                  <*> eitherMigrate caId
+                  -- ^ accWalletId
+
+instance Migrate V1.WalletId (V0.CId V0.Wal) where
+    eitherMigrate (V1.WalletId h) = pure $ V0.CId (V0.CHash h)
+
+--
+-- Following instances are friendly borrowed from @martoon's PR https://github.com/input-output-hk/cardano-sl/pull/2008
+-- TODO (jk): Use instances of his PR when it has been merged
+
+-- in old API 'V0.AccountId' supposed to carry both wallet id and derivation index
+instance Migrate (V1.WalletId, V1.AccountId) V0.AccountId where
+    eitherMigrate (walId, accId) =
+        V0.AccountId <$> eitherMigrate walId <*> pure accId
+
+instance Migrate V0.AccountId (V1.WalletId, V1.AccountId) where
+    eitherMigrate accId =
+        (,) <$> eitherMigrate (V0.aiWId accId) <*> pure (V0.aiIndex accId)
```

Comment:
If you want, this can be written using `TupleSections` also as:

```haskell
(,V0.aiIndex accId) <$> eitherMigrate (V0.aiWid accId)
```

But I think the "normal" applicative style is more elegant.

Reply:
Based on same PR #2008 So I think it's ok to leave it as it is for now. If needed we could improve/change it with next review of Kostja's PR.